evmjit/libevmjit-cpp/JitVM.cpp

```diff
@@ -18,27 +18,25 @@ namespace eth
 
 extern "C" void env_sload(); // fake declaration for linker symbol stripping workaround, see a call below
 
-bytesConstRef JitVM::go(ExtVMFace& _ext, OnOpFunc const& _onOp, uint64_t _step)
+bytesConstRef JitVM::go(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp, uint64_t _step)
 {
 	using namespace jit;
 
 	auto rejected = false;
 	// TODO: Rejecting transactions with gas limit > 2^63 can be used by attacker to take JIT out of scope
-	rejected |= m_gas > std::numeric_limits<decltype(m_data.gas)>::max(); // Do not accept requests with gas > 2^63 (int64 max)
+	rejected |= io_gas > std::numeric_limits<decltype(m_data.gas)>::max(); // Do not accept requests with gas > 2^63 (int64 max)
 	rejected |= _ext.gasPrice > std::numeric_limits<decltype(m_data.gasPrice)>::max();
 	rejected |= _ext.currentBlock.number > std::numeric_limits<decltype(m_data.number)>::max();
 	rejected |= _ext.currentBlock.timestamp > std::numeric_limits<decltype(m_data.timestamp)>::max();
 
 	if (rejected)
 	{
-		cwarn << "Execution rejected by EVM JIT (gas limit: " << m_gas << "), executing with interpreter";
-		m_fallbackVM = VMFactory::create(VMKind::Interpreter, m_gas);
-		auto&& output = m_fallbackVM->go(_ext, _onOp, _step);
-		m_gas = m_fallbackVM->gas(); // copy remaining gas, Executive expects it
-		return output;
+		cwarn << "Execution rejected by EVM JIT (gas limit: " << io_gas << "), executing with interpreter";
+		m_fallbackVM = VMFactory::create(VMKind::Interpreter);
+		return m_fallbackVM->go(io_gas, _ext, _onOp, _step);
 	}
 
-	m_data.gas 			= static_cast<decltype(m_data.gas)>(m_gas);
+	m_data.gas 			= static_cast<decltype(m_data.gas)>(io_gas);
```

Comment:
The positions of the `=` are in a funny place. I'd say either put them all normally (`a = b`) or align them all one under the other